creator-node/src/ipfsClient.js

```diff
@@ -0,0 +1,38 @@
+const ipfsClient = require('ipfs-http-client')
+const ipfsClientLatest = require('ipfs-http-client-latest')
+
+const config = require('./config')
+const { logger } = require('./logging')
+
+let ipfs
+let ipfsLatest
+
+// Make ipfs clients exportable to be used in rehydrate queue
+
+const initIPFS = async () => {
+  const ipfsAddr = config.get('ipfsHost')
+  if (!ipfsAddr) {
+    throw new Error('Must set ipfsAddr')
+  }
+  ipfs = ipfsClient(ipfsAddr, config.get('ipfsPort'))
+  ipfsLatest = ipfsClientLatest({ host: ipfsAddr, port: config.get('ipfsPort'), protocol: 'http' })
+
+  // initialize ipfs here
+  const identity = await ipfs.id()
+  // Pretty print the JSON obj with no filter fn (e.g. filter by string or number) and spacing of size 2
+  logger.info(`Current IPFS Peer ID: ${JSON.stringify(identity, null, 2)}`)
+
+  // init latest version of ipfs
+  const identityLatest = await ipfsLatest.id()
+  logger.info(`Current IPFS Peer ID (using latest version of ipfs client): ${JSON.stringify(identityLatest, null, 2)}`)
+
+  return { ipfs, ipfsLatest }
+}
+
+module.exports = (async function () {
```

Comment:
you generally don't want to have an async exports exported via module.exports. require doesn't respect that, especially since the order of dependencies and imports can be difficult to determine. so in these kinds of situations what you want to do is have the module.exports = { ipfs, ipfsLatest } and have another function like 
```
async function init () {
... 
} 
```
and export this as well and call this in the cnode init flow. strictly speaking this isn't necessary since the "init" just prints out the peer id of the ipfs node.

Reply:
I decided to immediately init the ipfs clients in `ipfsClient.js` since it is not heavy, and can be exported anywhere